aten/src/ATen/templates/NativeFunctions.h

```diff
@@ -11,3 +11,5 @@ namespace native {
 
 }
 }
+
+#include "ATen/native/NativeFunctions-inl.h"
```

Comment:
This is fine for this commit, but this won't scale well since everything that is templated has to be included everywhere. We probably need to add an optional `- implementation_header:` attribute that gets included only in the derived Type file when that template is actually being used in a file.

Reply:
We only have a limited set of types, can’t we just explicitly instantiate the template only for relevant types?

Reply:
We'd want to do that automatically, so we'd still need another attribute if we want users to be able to write the definitions in different files, which seems reasonable.  I.e. we'd need to know which instantiations to output to a file that gets included in the user definition file.  Since that requires more mechanism, I'm going to punt until this becomes an issue.